# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/WorkerScope.cpp
# Commit: 910dabb8545d
# Full Hash: 910dabb8545dfdea0c2b9385e347857042606bd7
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2023-06-07 03:30:33
# Regressor Bug: 1800659
# File Overlap Count: 1
# Description:
#   Bug 1800659 - P1 Adding log for Worker. r=asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D176915
# ==============================================================================

diff -r 9458874d5cf9 -r 910dabb8545d dom/workers/WorkerScope.cpp
--- a/dom/workers/WorkerScope.cpp	Tue Jun 06 06:28:14 2023 +0000
+++ b/dom/workers/WorkerScope.cpp	Tue Jun 06 06:36:50 2023 +0000
@@ -133,6 +133,13 @@
 
 namespace mozilla::dom {
 
+static mozilla::LazyLogModule sWorkerScopeLog("WorkerScope");
+
+#ifdef LOG
+#  undef LOG
+#endif
+#define LOG(args) MOZ_LOG(sWorkerScopeLog, LogLevel::Debug, args);
+
 class WorkerScriptTimeoutHandler final : public ScriptTimeoutHandler {
  public:
   NS_DECL_ISUPPORTS_INHERITED
@@ -239,6 +246,7 @@
       mClientSource(std::move(aClientSource)),
       mSerialEventTarget(aWorkerPrivate->HybridEventTarget()),
       mShouldResistFingerprinting(aShouldResistFingerprinting) {
+  LOG(("WorkerGlobalScopeBase::WorkerGlobalScopeBase [%p]", this));
   MOZ_ASSERT(mWorkerPrivate);
 #ifdef DEBUG
   mWorkerPrivate->AssertIsOnWorkerThread();
@@ -430,20 +438,12 @@
 WorkerGlobalScope::~WorkerGlobalScope() = default;
 
 void WorkerGlobalScope::NoteTerminating() {
+  LOG(("WorkerGlobalScope::NoteTerminating [%p]", this));
   if (IsDying()) {
     return;
   }
 
   StartDying();
-}
-
-void WorkerGlobalScope::NoteShuttingDown() {
-  MOZ_ASSERT(IsDying());
-
-  if (mNavigator) {
-    mNavigator->Invalidate();
-    mNavigator = nullptr;
-  }
 
   if (mPerformance) {
     RefPtr<PerformanceWorker> pw =
@@ -453,6 +453,16 @@
   }
 }
 
+void WorkerGlobalScope::NoteShuttingDown() {
+  MOZ_ASSERT(IsDying());
+  LOG(("WorkerGlobalScope::NoteShuttingDown [%p]", this));
+
+  if (mNavigator) {
+    mNavigator->Invalidate();
+    mNavigator = nullptr;
+  }
+}
+
 Crypto* WorkerGlobalScope::GetCrypto(ErrorResult& aError) {
   AssertIsOnWorkerThread();
 
